### PR TITLE
convert kind map to be representable as strings

### PR DESCRIPTION
**Pull request by schweitzpgi**

### CODE REVIEW

File: flang/include/flang/Optimizer/Support/KindMapping.h [hunk 1]
```diff
@@ -52,15 +52,24 @@ class KindMapping {
...
+  /// KindMapping constructor taking a `defs` argument to specify the default
+  /// kinds for intrinsic types. To set the default kinds, an ArrayRef of 6
+  /// KindTy must be passed. The kinds must be the given in the following order:
+  /// CHARACTER, COMPLEX, DOUBLE PRECISION, INTEGER, LOGICAL, and REAL.  The
+  /// kind map is read from command-line options, if given.
+  explicit KindMapping(mlir::MLIRContext *context, llvm::ArrayRef<KindTy> defs);
+  /// KindMapping constructor taking an optional `defs` argument to specify the
   /// default kinds for intrinsic types. To set the default kinds, an ArrayRef
   /// of 6 KindTy must be passed. The kinds must be the given in the following
   /// order: CHARACTER, COMPLEX, DOUBLE PRECISION, INTEGER, LOGICAL, and REAL.
```
Comment (vdonaldson) [inline flang/include/flang/Optimizer/Support/KindMapping.h:67]: What is the use case for DOUBLE PRECISION at this level?  Does it affect intrinsic call selection?

File: flang/include/flang/Optimizer/Support/KindMapping.h [hunk 2]
```diff
@@ -96,6 +111,16 @@ class KindMapping {
   KindTy defaultLogicalKind() const;
   KindTy defaultRealKind() const;
 
+  /// Get the default kinds as a string.
+  static constexpr const char *getDefaultKinds() { return "a1c4d8i4l4r4"; }
```
Comment (vdonaldson) [inline flang/include/flang/Optimizer/Support/KindMapping.h:115]: Maybe getDefaultDefaultKinds() ?

**Review by vdonaldson — COMMENTED** [review]
Will there be a test or example to illustrate how this works?

**Comment by schweitzpgi** [comment]
> Will there be a test or example to illustrate how this works?

Yes. It's in the unittest.

File: flang/include/flang/Optimizer/Support/KindMapping.h (hunk 1 above)
Comment (schweitzpgi) [reply flang/include/flang/Optimizer/Support/KindMapping.h:67]: Mainly it is for consistency with the vector we get from the front-end.

**Comment by vdonaldson** [comment]
> > Will there be a test or example to illustrate how this works?
> 
> Yes. It's in the unittest.

Is the new string representation ("a1c4d8i4l4r4") visible anywhere?

**Comment by schweitzpgi** [comment]
> > > Will there be a test or example to illustrate how this works?
> > 
> > 
> > Yes. It's in the unittest.
> 
> Is the new string representation ("a1c4d8i4l4r4") visible anywhere?

Not in the current test tools, no. This is just repackaging the sausage for upstreaming.

**Review by vdonaldson — APPROVED** [review]

**Pull request merged**